src/commands/createNewProject/dotnetSteps/DotnetRuntimeStep.ts

```diff
@@ -28,6 +28,14 @@ export class DotnetRuntimeStep extends AzureWizardPromptStep<IProjectWizardConte
         } else if (runtimes.length === 1) {
             // No need to prompt if it only supports one
             setWorkerRuntime(context, runtimes[0]);
+        } else if (context.targetFramework) {
```

Comment:
I think this case should be before the `runtimes.length === 1` case. If `runtimes` is `[netcoreapp2.1]` and `context.targetFramework` is `netcoreapp3.1`, I want it to throw your "Unrecognized target framework" error instead of using `2.1`